test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/EverythingOperationTests.cs

```diff
@@ -188,5 +188,73 @@ public async Task GivenMultipleInputParametersSpecified_WhenSomeInvalid_ThenInva
             FhirResponse<Bundle> secondBundle = await Client.SearchAsync(nextLink);
             ValidateBundle(secondBundle, Fixture.Observation);
         }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenPatientWithSeeAlsoLink_WhenRunningPatientEverything_ThenPatientEverythingShouldRunOnLink()
+        {
+            string searchUrl = $"Patient/{Fixture.PatientWithSeeAlsoLink.Id}/$everything";
+
+            FhirResponse<Bundle> firstBundle = await Client.SearchAsync(searchUrl);
+            ValidateBundle(firstBundle, Fixture.PatientWithSeeAlsoLink);
+
+            var nextLink = firstBundle.Resource.NextLink.ToString();
+            FhirResponse<Bundle> secondBundle = await Client.SearchAsync(nextLink);
+            Assert.Empty(secondBundle.Resource.Entry);
+
+            nextLink = secondBundle.Resource.NextLink.ToString();
+            FhirResponse<Bundle> thirdBundle = await Client.SearchAsync(nextLink);
+            ValidateBundle(thirdBundle, Fixture.PatientReferencedByLink);
+
+            nextLink = thirdBundle.Resource.NextLink.ToString();
+            FhirResponse<Bundle> fourthBundle = await Client.SearchAsync(nextLink);
+
+            // All the patients that have links to the "patient-reference" patient will be in its patient compartment
+            ValidateBundle(fourthBundle, Fixture.PatientWithSeeAlsoLink, Fixture.PatientWithReplacedByLink, Fixture.PatientWithReferLink, Fixture.PatientWithReplacesLink);
+        }
```

Comment:
Another case is that when patient A points to patient B and a resource C points to both patient A and B, C will be returned twice in the response as well, one in the compartment search of A and the other in the compartment search of B.

Reply:
Really good question. In the case where Patient A points to Patient B, I had it designed to consider the patient $everything operation on Patient A separately from the patient $everything operation on Patient B. I'll reach out to @CaitlinV39!

Reply:
As Boya mentioned this is why we didn't do it in the first place lol. It would be best if when we do the second everything call if the patient A isn't returned twice. Is that possible. On the second example from Boya, I agree that it would be great to ensure that no duplicates are returned but since this isn't all returned in one query for the post-processing, I can just document that there could be duplicates. 

This is making me wonder if having a flag to not follow the links would be valuable though. So you could do Patient/A/$everything where you could say you want to follow the first level links or you could pass a parameter to not follow the first level links. The first option would ensure no duplicates and the second would give more data, with a small chance of duplicates. If we do that, then I'm probably okay with duplicate Patient A as well. In SQL we can do some refactoring and we might be able to optimize this behavior (not for now, just a thought fo the future).

Reply:
If we just want to make sure that the patient itself is not returned multiple times, it can be achieved by adding a post-process. This is because if the patient itself is in scope for response, it will always be returned in first page, so we can just filter it out in later pages. But this does not apply for other resources. For example, if Patient A and B "seealso" link to each other, when "Patient/A/$everything" is called, though A is no longer duplicate after adding the post-process, B is still duplicate, one in A's compartment and the other in the "seealso" link.

Reply:
Thanks for all the thought on this!

RE:
> Another case is that when patient A points to patient B and a resource C points to both patient A and B, C will be returned twice in the response as well, one in the compartment search of A and the other in the compartment search of B.

I had a discussion with @CaitlinV39 and @brendankowitz and confirmed that we will allow duplicates in this scenario

RE:
> In this case where patient A points to patient B, if "Patient/A/$everything" is called, A will be returned twice in the response. It first appears when searching itself. It then appears when searching the compartment of B in the "seealso" link. In previous discussion of patient $everything, we are not going to return duplicate resources in the response. Are we expecting to change this in the update?

I removed the parent link from the `seealso` patient's compartment in c690d1d.

Reply:
Added a comment about this in a8f5b52.